src/H5Dint.c

```diff
@@ -2256,7 +2256,8 @@ H5D_nameof(const H5D_t *dataset)
     /* Use FUNC_ENTER_NOAPI_NOINIT_NOERR here to avoid performance issues */
     FUNC_ENTER_NOAPI_NOINIT_NOERR
 
-    FUNC_LEAVE_NOAPI(dataset ? &(dataset->path) : NULL)
+    /* TODO: should not cast away const! */
+    FUNC_LEAVE_NOAPI(dataset ? (H5G_name_t *)&(dataset->path) : NULL)
```

Comment:
Please don't cast away const, it's hiding the underlying design issue, which should be fixed.

Reply:
Certainly fixing the underlying design issue would be preferable.

The cons of this change:
- a useful compiler warning has been downgraded to a TODO comment

The pros of this change:
- there's no real change. an implicit cast is now explicit. codegen will be identical.
- the codebase is now clean of -Wincompatible-pointer-types-discards-qualifiers warnings and the flag could now be enabled in your CI, thus preventing any ***new*** such warnings being introduced.

For me, this latter point outweighs the cons.

Reply:
Yes, I understand the pros & cons. :-)  But the cons are much more important, I believe, since the underlying problem has not been resolved and probably never will be, since it's completely invisible to developers.  I understand the benefit to the overall codebase by clearing the codebase of the type of warning, but this sort of fix just encourages other developers to "fix" the warning by just casting away const also, instead of actually fixing the underlying design problem.  It generates a false sense of "security" since there's no warning, but the underlying problem still remains active.

Reply:
OK, I'll drop this commit then.